src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java

```diff
@@ -33,31 +33,89 @@
 import org.opensearch.common.xcontent.LoggingDeprecationHandler;
 import org.opensearch.index.IndexNotFoundException;
 import org.opensearch.index.engine.DocumentMissingException;
+import org.opensearch.index.engine.Engine;
 import org.opensearch.index.engine.VersionConflictEngineException;
 import org.opensearch.index.seqno.SequenceNumbers;
+import org.opensearch.index.shard.IndexingOperationListener;
+import org.opensearch.index.shard.ShardId;
 import org.opensearch.jobscheduler.model.JobDetails;
 
 import java.nio.charset.StandardCharsets;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
-public class JobDetailsService {
+public class JobDetailsService implements IndexingOperationListener {
 
     private static final Logger logger = LogManager.getLogger(JobDetailsService.class);
-    private static final String JOB_DETAILS_INDEX_NAME = ".opensearch-plugins-job-details";
+    public static final String JOB_DETAILS_INDEX_NAME = ".opensearch-plugins-job-details";
     private static final String PLUGINS_JOB_DETAILS_MAPPING_FILE = "/mappings/opensearch_plugins_job_details.json";
 
     public static Long TIME_OUT_FOR_REQUEST = 10L;
     private final Client client;
     private final ClusterService clusterService;
+    private Set<String> indicesToListen;
+    private final ConcurrentMap<String, JobDetails> indexToJobDetails;
 
-    public JobDetailsService(final Client client, final ClusterService clusterService) {
+    public JobDetailsService(final Client client, final ClusterService clusterService, Set<String> indicesToListen) {
         this.client = client;
         this.clusterService = clusterService;
+        this.indicesToListen = indicesToListen;
+        this.indexToJobDetails = new ConcurrentHashMap<>();
     }
 
     public boolean jobDetailsIndexExist() {
         return clusterService.state().routingTable().hasIndex(JOB_DETAILS_INDEX_NAME);
     }
 
+    public ConcurrentMap<String, JobDetails> getIndexToJobDetails() {
+        return this.indexToJobDetails;
```

Comment:
Implement the return of object of indexToJobdetails in below manner.

private static class InstanceHolder {
    private static final YourObject instance = new YourObject();
}

public static YourObject getInstance() {
    return InstanceHolder.instance;
}
This solution takes advantage of the Java memory model's guarantees about class initialization to ensure thread safety. Each class can only be loaded once, and it will only be loaded when it is needed. That means that the first time getInstance is called, InstanceHolder will be loaded and instance will be created, and since this is controlled by ClassLoaders, no additional synchronization is necessary.

In the case above it would look like 

JobDetailsService.indexToJobDetails

Reply:
Sure I'll go ahead and add this, I'll keep the method name `getIndexToJobDetails`, appending Map isn't necessary